src/main/java/InfrastructureManager/Modules/NetworkStructure/Network.java

```diff
@@ -237,50 +236,222 @@ else if(update == Network.Update.DELETE) {
 	}
 
 	/**
-	 * Getter function
-	 * @return applicationDeviceList
+	 * Getter function to get a particular ApplicationInstanceDeviceRelation from applicationDeviceList
+	 * @param id
+	 * @return ApplicationInstanceDeviceRelation
 	 */
-	public List<ApplicationInstanceDeviceRelation> getApplicationDeviceList() {
-		return applicationDeviceList;
+	public ApplicationInstanceDeviceRelation getApplicationInstanceDeviceRelationFromList(String id)	{
+		int idx=0;
+		for (Object obj : this.applicationDeviceList) {
+			ApplicationInstanceDeviceRelation applicationDevice = (ApplicationInstanceDeviceRelation)obj;
+			if ((applicationDevice.getApplication().getApplicationType().getApplicationId().equals(id)) || applicationDevice.getApplicationDevice().getDeviceId().equals(id))
+			{
+				idx = this.applicationDeviceList.indexOf(obj);
+			}
+		}
+		return this.applicationDeviceList.get(idx);
 	}
 
 	/**
-	 * Getter function
-	 * @return locationConnectionList
+	 * Function to update a particular applicationDevice in applicationDeviceList
+	 * @param applicationDevice
+	 * @param UpdateType // ADD or DELETE
 	 */
-	public List<LocationConnectionRelation> getLocationConnectionList() {
-		return locationConnectionList;
+	public void updateapplicationDeviceList(ApplicationInstanceDeviceRelation applicationDevice,Network.Update update){
+		if (update == Network.Update.ADD)
```

Comment:
Now is a good time to actually separate the add and delete functions. There should be one function just for adding and another one just for deleting. This applies to the whole class not just this function.